activerecord/lib/active_record/enum.rb

```diff
@@ -225,6 +225,8 @@ def detect_enum_conflict!(enum_name, method_name, klass_method = false)
           raise_conflict_error(enum_name, method_name)
         elsif !klass_method && method_defined_within?(method_name, _enum_methods_module, Module)
           raise_conflict_error(enum_name, method_name, source: "another enum")
+        elsif !klass_method && method_defined_within?(method_name, Relation)
```

Comment:
When this method is called to detect scope name conflicts, `klass_method` is true:
https://github.com/rails/rails/blob/v5.1.4/activerecord/lib/active_record/enum.rb#L197

A scope is a class method on the model, even if we sometimes access it as an instance method on `Relation`.

Reply:
Ok, So why did the test pass? i

Reply:
Ohh I see, can you show me where is the code that allows accessing the scope via the relation instace?

Reply:
https://github.com/rails/rails/blob/v5.1.4/activerecord/lib/active_record/relation/delegation.rb#L90